gnuradio-runtime/include/gnuradio/basic_block.h

```diff
@@ -28,6 +28,28 @@
 
 namespace gr {
 
+class msg_queue_comparator
+{
+private:
+    const pmt::pmt_t d_system_port = pmt::intern("system");
+
+public:
+    bool operator()(pmt::pmt_t const& queue_key1, pmt::pmt_t const& queue_key2) const
+    {
+        bool is_system1 = pmt::eqv(queue_key1, d_system_port);
+        bool is_system2 = pmt::eqv(queue_key2, d_system_port);
+
+        if (is_system1 && is_system2)
```

Comment:
This is equivalent to
```
if (is_system2) return false;
else if (is_system1) return true;
else ...
```
but perhaps it's more obvious the way you have it.

Reply:
I left it the long way since it seemed a bit more understandable. I'll leave it unless there's a strong preference to shorten it.

Reply:
Nah, the compiler will take care of it anyway.

Reply:
I suggest to do a quick check with https://godbolt.org/
You never know what your compiler does.

Reply:
The code is actually quite a bit simpler the other way, since the `is_system1` and `is_system2` don't need to be separately defined. I'll rewrite it.